wagtail_transfer/richtext.py

```diff
@@ -0,0 +1,85 @@
+import re
+
+from wagtail.core.rich_text.feature_registry import FeatureRegistry
+from wagtail.core.rich_text.rewriters import extract_attrs
+
+from .models import get_base_model
+
+features = FeatureRegistry()
+
+FIND_A_TAG = re.compile(r'<a(\b[^>]*)>')
+FIND_EMBED_TAG = re.compile(r'<embed(\b[^>]*)/>')
+FIND_ID = re.compile(r'id="([^"]*)"')
+
+
+class RichTextReferenceHandler:
+    """
+    Handles updating ids and retrieving object references for <tag type_attribute="foo" id="my_id" /> tags representing references within rich text. Tags are found using a regex tag_matcher, and their
+    specific handler (eg PageLinkHandler) is looked up using a dict, handlers, which maps the value of type_attribute (where type_attribute might be emebedtype)
+    to a specific handler.
+    """
+    def __init__(self, handlers, tag_matcher, type_attribute, destination_ids_by_source={}):
+        self.handlers = handlers
+        self.tag_matcher = tag_matcher
+        self.type_attribute = type_attribute
+        self.destination_ids_by_source = destination_ids_by_source
+
+    def update_tag_id(self, match):
+        # Updates a specific tag's id from source to destination Wagtail instance
+        attrs = extract_attrs(match.group(1))
+        try:
+            handler = self.handlers[attrs[self.type_attribute]]
+            target_model = get_base_model(handler.get_model())
+            new_id = self.destination_ids_by_source.get((target_model, int(attrs['id'])), int(attrs['id']))
```

Comment:
Technically we shouldn't be assuming the presence of an `id` attribute - it should be internal to the linkhandler/embedhandler how it chooses to translate the attribute dict into an object instance. In practice, this probably doesn't matter because:

1) the only current exception is `embedtype="media"`, which has a URL instead of an ID, and for messy reasons that I don't need to go into here, skipping over those is the right thing to do in this case (and that's exactly what the `try / except KeyError` will do)
2) the only way to avoid this method dealing with id attributes is to delegate the job of replacing IDs to the linkhandler/embedhandler, and inventing new methods for that purpose is really outside the scope of this package.